armi/bookkeeping/db/tests/test_comparedb3.py

```diff
@@ -158,6 +165,78 @@ def test_compareDatabaseSim(self):
         self.assertEqual(len(diffs.diffs), 456)
         self.assertEqual(diffs.nDiffs(), 3)
 
+    def test_diffSpecialData(self):
+        dr = DiffResults(0.01)
+
+        fileName = "test_diffSpecialData.txt"
+        with OutputWriter(fileName) as out:
+            # spin up one example H5 Dataset
+            f1 = h5py.File("test_diffSpecialData1.hdf5", "w")
+            a1 = np.arange(100, dtype="<f8")
+            refData = f1.create_dataset("numberDensities", data=a1)
+            refData.attrs["1"] = 1
+            refData.attrs["2"] = 22
+            refData.attrs["numDens"] = a1
+
+            # spin up an identical example H5 Dataset
+            f2 = h5py.File("test_diffSpecialData2.hdf5", "w")
+            srcData = f2.create_dataset("numberDensities", data=a1)
+            srcData.attrs["1"] = 1
+            srcData.attrs["2"] = 22
+            srcData.attrs["numDens"] = a1
+
+            # there should be no difference
+            _diffSpecialData(refData, srcData, out, dr)
+            self.assertEqual(dr.nDiffs(), 0)
+
+            # spin up a different size example H5 Dataset
+            f3 = h5py.File("test_diffSpecialData3.hdf5", "w")
+            a2 = np.arange(90, dtype="<f8")
+            srcData3 = f3.create_dataset("numberDensities", data=a2)
+            srcData3.attrs["1"] = 1
+            srcData3.attrs["2"] = 22
+            srcData3.attrs["numDens"] = a2
+
+            # there should be no difference
+            with mockRunLogs.BufferLog() as mock:
+                _diffSpecialData(refData, srcData3, out, dr)
+                self.assertEqual(dr.nDiffs(), 0)
```

Comment:
Now that I'm thinking about it, it's weird that this doesn't first add a diff to the count. Not proposing that the code change, just mentioning it is counterintuitive (in hindsight)

Reply:
Yeah, what we get here is a logged error, and the method executes.

Essentially, the logged error says it can't _do_ any comparison. Which means the comparison count it lower than it should be. But... you're not wrong.

Reply:
true true. The error should be enough